integration-libs/opf/components/opf-verify-payment/opf-verify-payment.component.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  RoutingService,
+} from '@spartacus/core';
+import {
+  KeyValuePair,
+  OpfCheckoutFacade,
+  OpfConfig,
+  OpfVerifyPaymentResponse,
+} from '@spartacus/opf/root';
+import { OrderFacade } from '@spartacus/order/root';
+import { Subscription, throwError } from 'rxjs';
+import { switchMap, take } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-opf-verify-payment',
+  templateUrl: './opf-verify-payment.component.html',
+})
+export class OpfVerifyPaymentComponent implements OnInit, OnDestroy {
+  subscription?: Subscription;
+
+  constructor(
+    protected route: ActivatedRoute,
+    protected routingService: RoutingService,
+    protected orderFacade: OrderFacade,
+    protected opfCheckoutService: OpfCheckoutFacade,
+    protected config: OpfConfig,
+    protected globalMessageService: GlobalMessageService
+  ) {}
+
+  ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  ngOnInit(): void {
+    this.subscription = this.route.queryParams
+      .pipe(
+        take(1),
+        switchMap((params) => {
+          if (!params) return throwError('No params');
+
+          const list: KeyValuePair[] = Object.entries(params).map((pair) => {
```

Comment:
Can we have this part as a separate method, and ideally in a service?

Reply:
created 'OpfUrlHandlerService' which has generic methods
convertParamsToKeyValuePairs
findFromKeyValuePairs
isIncludedInPath
getDomainUrl
This service is in Components folder, still need some work to move it to Services folder as it throws an error in runtime